src/mono/mono/mini/mini-amd64.c

```diff
@@ -2544,7 +2544,7 @@ mono_arch_emit_setret (MonoCompile *cfg, MonoMethod *method, MonoInst *val)
 typedef struct {
 	MonoMethodSignature *sig;
 	CallInfo *cinfo;
-	int nstack_args, nullable_area;
+	int nstack_args;
```

Comment:
@vargaz This PR removes special handling of nullables from the runtime invoke paths. The corresponding change in runtime invoke wrapper is made in `emit_invoke_call` in commit `[mono] Implement new version of InternalInvoke icall`. There is a risk for breaking changes with embedders but I don't think it is worthwhile to maintain separate versions of invoke wrappers and code.

Reply:
> There is a risk for breaking changes with embedders 

How does this affect embedders? Do they need to manually box nullable arguments before calling `mono_runtime_invoke`? That seems like a pretty big change

Reply:
@lambdageek So consider the case where we do a runtime invoke into a method receiving a nullable argument. This method will expect to receive a nullable structure. The old Invoke API would receive an array of objects, so the caller of the Invoke API, when converting the argument into an object it will result a boxed value type. The invoke wrapper used to receive this boxed value type and it would unbox it into a nullable structure to pass it along. The new API would pass directly a ref to the nullable structure to the `InternalInvoke`, so the runtime invoke wrapper doesn't need to do anything else.

Now considering the case for embedders, this behavior of unboxing a valuetype into a nullable structure doesn't really seem to be documented and I figure it was just something for us to easily support the InternalInvoke behavior. The documentation of `mono_runtime_invoke` states that it should receive a `MonoObject*` for objects and a pointer to the value type for valuetypes, so this implementation still follows the documentation for the invokes. It just stops supporting, the apparently undocumented, behavior of allowing to pass a boxed valuetype as an argument for a method receiving a nullable.

Reply:
I think we could also do the conversion from boxed vt to unboxed nullable only for `mono_runtime_invoke` and `mono_runtime_invoke_array`, just to preserve behavior

Reply:
That should work.

Reply:
So the nullable behavior is going to be handled in the invoke wrapper, so the dyn call code doesn't need these special cases anymore ?